csrc/multidevice/communication.cpp

```diff
@@ -475,10 +475,10 @@ c10::intrusive_ptr<c10d::Work> postReduceScatter(
   // user's input tensors to an internal source buffer. It is therefore always
   // preferable to use _reduce_scatter_base (which does not perform any extra
   // copy) when the tensors are stored contiguously (i.e., when
-  // scattered_axis==0). Note however than only nccl supports
+  // scattered_axis==1). Note however than only nccl supports
   // _reduce_scatter_base, not ucc.
 #if defined(NVFUSER_DISTRIBUTED) && defined(USE_C10D_NCCL)
-  if (scattered_axis == 0 &&
+  if (scattered_axis == 1 && communication->out()->axis(0)->isReduction() &&
```

Comment:
Will this change cause regression for cases like the following? 

```
auto* tv0 = makeConcreteContigTensor({-1, d});
auto* tv1 = sum(tv0, {1});

tv0->axis(-1)->parallelize(DIDx);
tv1->outer_split(0, d);
tv1->axis(0)->parallelize(DIDx);
```

The scattered axis is 0 and it should be able to use _reduce_scatter_base.

Reply:
Also note scattered_axis is [computed using](https://github.com/NVIDIA/Fuser/blob/01d89cff65d5e5dd30d8f54d55e0a8c340311468/csrc/host_ir/lower.cpp#L283) getShardedLogicalAxis, which [ignores reduction](https://github.com/NVIDIA/Fuser/blob/01d89cff65d5e5dd30d8f54d55e0a8c340311468/csrc/multidevice/utils.cpp#L157) dimensions already.

Reply:
> What is ambiguous in your example though is the contiguity of tv0

I thought makeConcreteContigTensor says that. What am I missing?

Reply:
> > What is ambiguous in your example though is the contiguity of tv0
> 
> I thought makeConcreteContigTensor says that. What am I missing?

Yes, the tensor you define is contiguous, but this is not currently checked explicitely for the condition to use `reduce_scatter_base` which merely looks at the value of scattered_axis.